Documentation/coreos.rst

```diff
@@ -0,0 +1,376 @@
+.. _coreos_gs_guide:
+
+Getting Started Guide for CoreOS
+==================================
+
+This document serves as a guide to get Cilium up-and-running on CoreOS.
+
+If you haven't read the :ref:`intro` yet, we encourage you to do that first.
+
+For an introduction about how to setup Kubernetes with CoreOS, we recommend
+following along with the `Core OS guide <https://coreos.com/kubernetes/docs/latest/getting-started.html>`_.
+The CoreOS guide serves as a reference to setting up Kubernetes components (e.g., kube-apiserver, kube-scheduler, certificates, etc.), while this page focuses specifically on the modifications needed to get Cilium functioning with CoreOS.
+
+The best way to get help if you get stuck is to ask a question on the `Cilium 
+Slack channel <https://cilium.herokuapp.com>`_. With Cilium contributors
+across the globe, there is almost always someone available to help.
+
+
+Minimum Requirements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Make sure you understand the various `deployment options <https://coreos.com/kubernetes/docs/latest/getting-started.html#deployment-options>`_.
+* *etcd*
+    * ``etcd`` must be at version ``>= 3.1.0``. To set up etcd, follow `the CoreOS instructions for setting up an etcd cluster <https://coreos.com/kubernetes/docs/latest/getting-started.html#deploy-etcd-cluster>`_.
+* *kubectl*
+    * It is also required that you install ``kubectl`` version ``>= 1.6.4`` as described in the `Kubernetes Docs <https://kubernetes.io/docs/tasks/tools/install-kubectl/>`_.
+
+Setup Master Nodes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Step 1: Create TLS Assets
+--------------------------------
+Follow the `CoreOS instructions for creating TLS assets on master nodes <https://coreos.com/kubernetes/docs/latest/getting-started.html#generate-kubernetes-tls-assets>`_ and `install the TLS assets needed for the master nodes <https://coreos.com/kubernetes/docs/latest/deploy-master.html#tls-assets>`_ for secure communication with the kube-apiserver.
+
+
+Step 2: Setup Kubelet
+--------------------------------
+
+On CoreOS, Kubelet (the Kubernetes agent that runs on each node, more info `here <https://kubernetes.io/docs/admin/kubelet/>`_) runs as a container. In order to securely communicate with the API server, kubelet uses the TLS assets we generated as part of Step 1.
+
+Master nodes are usually not scheduled to run workloads, so we provide the ``-register-schedulable=false`` in the example YAML excerpt below. This ensures that workloads are only scheduled on worker nodes.
+
+Since we are setting up Kubelet to use Cilium, we want to configure its networking to utilize CNI (Container Networking Interface). This ensures that each pod that is created can communicate with one another within the cluster with Cilium networking configured.
+
+* Replace ${ADVERTISE_IP} with this node's publicly routable IP.
+* Replace ${DNS_SERVICE_IP}
```

Comment:
Replace with what? Let's assume this is the first time the user is seeing this so also tell the user that should be an IP from the service-cidr (SERVICE_CLUSTER_IP_RANGE) set bellow